app/Classes/Intervention/Transform.php

```diff
@@ -19,18 +22,72 @@ class Transform implements TransformInterface
      * @param array|null $transformations
      *
      * @return string Binary string of the image.
+     */
+    public function transform(string $pathToOriginalImage, ?array $transformations = null): string
+    {
+        $fileHandle = $this->getOriginalFileStream($pathToOriginalImage);
+        $mimeType = mime_content_type($fileHandle);
+        $fileData = stream_get_contents($fileHandle);
+
+        return match ($mimeType) {
+            'application/pdf' => $this->pdfToImage($fileData, $transformations),
+            default => $this->applyTransformations($fileData, $transformations),
+        };
+    }
+
+    /**
+     * @param string $fileData
+     * @param array|null $transformations
+     * @return string
+     * @throws DocumentPageDoesNotExistException
+     */
+    protected function pdfToImage(string $fileData, ?array $transformations = null): string
+    {
+        // We need a local file for Imagick to be able to access only the requested page.
+        $tempFile = tempnam(sys_get_temp_dir(), 'transmorpher');
+        file_put_contents($tempFile, $fileData);
+
+        try {
+            $imagick = new Imagick();
+
+            // 300 DPI, might make this a transformation
```

Comment:
?

Reply:
DPI can now be specified as transformation.